src/num_entries.rs

```diff
@@ -0,0 +1,93 @@
+//! Trait to report object size as the number of entries.
+
+use impl_trait_for_tuples::impl_for_tuples;
+use std::rc::Rc;
+
+/// Trait to report object size as the number of entries.
+///
+/// Scalars have size 1.  Container (e.g., vector, map) size is
+/// the sum of sizes of its elements.  If elements have constant size,
+/// container size can be efficiently measured by multiplying the
+/// number of elements by the size of each element.
+pub trait NumEntries {
+    /// Returns the number of entries in `self`.
+    fn num_entries(&self) -> usize;
+
+    /// Returns `Some(n)` if `Self` has constant size or `None` otherwise.
+    fn const_num_entries() -> Option<usize>;
```

Comment:
If you want to make this an actual const you can just make it an associated constant,
```suggestion
    /// Is `Some(n)` if `Self` has constant size and `None` otherwise.
    const STATIC_ENTRIES: Option<usize> = None;
```